Converter merge

DrDudelsack opened the pull request:
Implementation of the converter written in Java

DrDudelsack committed:
Uploaded core converter files

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create blank

DrDudelsack committed:
Create 50_convertedBloodlines.txt

DrDudelsack committed:
Create 00_bookmarks.txt

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create 00_cultures.txt

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create 50_romeCultures

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Rename 50_romeCultures to 50_romeCultures.txt

DrDudelsack committed:
Add files via upload

DrDudelsack committed:
Create culture_conversion_events.txt

DrDudelsack committed:
Create blank

DrDudelsack committed:
Create readme.txt

DrDudelsack committed:
Create 50_convertedReligions.txt

DrDudelsack committed:
Create k_TAG.txt

DrDudelsack committed:
Create readme.txt

DrDudelsack committed:
Delete Main.cpp

DrDudelsack committed:
Update TestQ.java

DrDudelsack committed:
Added compatibility with the Paradox Mapper tool's province mappings

DrDudelsack committed:
Added compatibility with the Paradox Mapper tool's province mappings

DrDudelsack committed:
Removed commented code and unused functions

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+        
+        try {
+        
+        while (flag == 0) {
```
You should be able to do
```
while (!vmm.equals(keyword)) {
  vmm = scnr.nextLine();
}
```

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+        Scanner scnr= new Scanner(fileIn);
+        
+        FileOutputStream fileOut= new FileOutputStream(type);
+        PrintWriter out = new PrintWriter(fileOut);   
```
Move these to just before the loop where you use them.

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+        }
+        }
+        
+        while (flag == 1) {
```
As with the other loop, you should just be able to do:
```
while (!vmm.equals(endword)) {
  out.println(vmm);
  vmm = scnr.nextLine();
}
out.println(vmm);
```

Drop the last line if you don't want the endword output with the rest.

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+    
+    public static void tempCreate(String save, String keyword, String endword, String type) throws IOException
+    {
+        int flag = 0;
```
With the changes suggested below, this variable is no longer necessary and can be deleted.

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+public class TempFiles
+{
+    
+    private int x;
```
This doesn't look to be used anywhere, so delete it.

Idhrendur commented on ImperatorToCk2/TempFiles.java:
```diff
@@ -0,0 +1,65 @@
+
+
+    
+    public static void tempCreate(String save, String keyword, String endword, String type) throws IOException
```
Try ```tempFile``` instead of ```type```

Idhrendur replied:
The usual convention for function naming is verbNoun, so createTemp() would be better, and createTempFile() even better than that.

As well, temp tells me about the lifetime of the file, but not what it is or what its for. You might also consider 'PartialFile', 'SaveSection' or some other adjective or adjective phrase.

DrDudelsack committed:
Ran classes through a code formatter

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+ */
+public class Directories
+{
+    private int x;
```
This doesn't look to be used anywhere. Delete it.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+    
+    public static int modFolders (String Dir,String modName ) throws IOException
+   {
+     //Nessicery to create all of the folders required for the mod
```
Nessicery -> Necessary

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+     //Nessicery to create all of the folders required for the mod
+     int aqv = 0;
+     boolean  endOrNot2 = true;
+     String VM = "\\";
```
"/" will work better on non-Windows systems

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+    public static int modFolders (String Dir,String modName ) throws IOException
+   {
+     //Nessicery to create all of the folders required for the mod
+     int aqv = 0;
```
This isn't used except to return. Remove the variable (and also the return).

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+   {
+     //Nessicery to create all of the folders required for the mod
+     int aqv = 0;
+     boolean  endOrNot2 = true;
```
This isn't used, remove.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+     String VM = "\\";
+     VM = VM.substring(0);
+     String mainModFolder = Dir+VM+modName;
+     File f = new File(mainModFolder);
```
Each of the Files should have a meaningful name: commonFolder, eventsFolder, etc.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+     File f20 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy"+VM+"common");
+        f20.mkdir();
+        
+     File f21 = new File(mainModFolder+VM+"eu4_converter"+VM+"copy"+VM+"common"+VM+"religions");
```
This is a hard to read, and a little inefficient due to Java's string concatenation. Instead do:
```
mainModFolder + "/eu4_converter/copy/common/religions"
```

Update all the others similarly.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+    private int x;
+
+    
+    public static int modFolders (String Dir,String modName ) throws IOException
```
Instead of Dir, I'd suggest CK2ModLocation

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+    return aqv;
+    }
+    
+    public static String descriptors(String Dir, String modName, String Dir2) throws IOException 
```
The return type isn't meaningful or used, so change to void.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+import java.util.Scanner;
```
Looks like this file was missed by the formatter. Give it a quick format.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+        char VMq = '"';
+        String VN = "//";
+       VN = VN.substring(0);
+        VM = VM.substring(0);
```
VM, VMq, and VN are all unnecessary. Just include the characters directly in the strings below.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+        String VN = "//";
+       VN = VN.substring(0);
+        VM = VM.substring(0);
+        int aqq = 1;
```
aqq isn't used anywhere. Delete it.

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+       VN = VN.substring(0);
+        VM = VM.substring(0);
+        int aqq = 1;
+        String mainModFolder = Dir+VM+modName;
```
mainModFolder -> modFile

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+        VM = VM.substring(0);
+        int aqq = 1;
+        String mainModFolder = Dir+VM+modName;
+        String dotModFolder = Dir2+VN+modName;
```
dotModFolder -> descriptorFile

Idhrendur commented on ImperatorToCk2/Directories.java:
```diff
@@ -0,0 +1,130 @@
+        VM = VM.substring(0);
+        int aqq = 1;
+        String mainModFolder = Dir+VM+modName;
+        String dotModFolder = Dir2+VN+modName;
```
No need for Dir2, you can instead do 
```
String descriptorFile = Dir + "/" + modname + "/descriptor.mod";
```

Idhrendur replied:
Also, looks like you aren't currently outputting descriptor.mod. You should do so.

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Accidently copied the formatted file into the wrong folder

DrDudelsack committed:
Moved formatted classes

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    private int x;
+
+    
+    public static int output(String source, String destination) throws IOException
```
It looks like this does a line-by-line copy of files. Instead of this, use java.nio.Files.copy to just directly copy files: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#copy(java.nio.file.Path,%20java.io.OutputStream)

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        Importer importer = new Importer();
+
+        ck2CultureInfo = importer.importCultList("cultureConversion.txt",irCulture)[1];
```
Just return the value instead of putting it in a variable. Then remove the now-unused variable.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    {
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
```
This isn't used, so remove.

Idhrendur commented on ImperatorToCk2/Importer.java:
```diff
@@ -0,0 +1,616 @@
+    
+    }
+   
+    public static String[] importCultList (String name, String provIDnum) throws IOException
```
As this is used for more than just cultures, rename it to a more general description.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    public static String titleCreationCommon(String irTAG, String irColor, String isRepublic,String capital,String rank, String Directory) throws IOException
+    {
+
+        String tab = "	";
```
Unnecessary, just use \t

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String titleCreationCommon(String irTAG, String irColor, String isRepublic,String capital,String rank, String Directory) throws IOException
```
Directory -> modDirectory

Idhrendur replied:
Use a boolean for isRepublic

Idhrendur replied:
titleCreationCommon -> createTitleInCommon
or something similar

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String tab = "	";
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        Directory = Directory + VM + "common" + VM + "landed_titles";
```
Directory -> landedTitlesDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String tab = "	";
+        String VM = "\\"; 
+        VM = VM.substring(0);
```
As elsewhere, just use "/" in the strings or expressions themselves.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return irColor;
+    }
+
+    public static String titleCreation(String irTAG, String irKING, String irCOLOR, String isRepublic, String capital,String rank,String liege,
```
Use a boolean instead of a String for isRepublic

Idhrendur replied:
The return type doesn't seem significant, so make this function return void.

Idhrendur replied:
titleCreation -> createTitle

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -17,344 +17,320 @@
-        
-        
+
         String VM = "\\"; 
```
As elsewhere, leave this out and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println (tab+"color={ "+irColor+" }");
+        out.println (tab+"color2={ "+irColor+" }");
+
+        if (capital != "none") { //governorships don't have set capitals
```
```
if (!capital.equals("none") {
```

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String irKING100 = "9"+irKING;
+
+        String tab = "	";
```
As elsewhere, delete this and just use "\t" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String date2 = "1066.9.15";
+
+        out.println (date1+"={");
+        if (liege != "no_liege") {
```
```
if (!liege.equals("no_liege")) {
```

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String tab = "	";
+
+        titleCreationCommon(irTAG,irCOLOR,isRepublic,capital,rank,Directory);
+        Directory = Directory + VM + "history" + VM + "titles";
```
Instead of modifying this, just have the full path there when creating the FileOutputStream

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        titleCreationCommon(irTAG,irCOLOR,isRepublic,capital,rank,Directory);
+        Directory = Directory + VM + "history" + VM + "titles";
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
```
This is only used for the return and unmodified, so delete it.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        titleCreationCommon(irTAG,irCOLOR,isRepublic,capital,rank,Directory);
+        Directory = Directory + VM + "history" + VM + "titles";
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
+        Importer importer = new Importer();
```
This isn't use in this function, so delete.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        FileOutputStream fileOut= new FileOutputStream(Directory + VM + rank+"_" + irTAG + ".txt");
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
```
Unused, delete.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        int flag = 0;
+
+        String date1 = "100.1.1";
```
This is used several times in this file, but redifined in each function. Instead, make it a private static final member variable of the class.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        int flag = 0;
+
+        String date1 = "100.1.1";
+        String date2 = "1066.9.15";
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        out.println (date1+"={");
+        if (liege != "no_liege") {
+            out.println (tab+"liege="+"k_"+liege);
```
You can combine some of these strings:
```
out.println(tab + "liege = k_" + liege);

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println (date1+"={");
+        if (liege != "no_liege") {
+            out.println (tab+"liege="+"k_"+liege);
+            out.println (tab+"de_jure_liege="+"k_"+liege);
```
Ditto.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println ();
+
+        out.println (date2+"={");
+        if (liege != "no_liege") {
```
Same as above

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        out.println (date2+"={");
+        if (liege != "no_liege") {
+            out.println (tab+"liege="+"k_"+liege);
```
Same as above

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println (date2+"={");
+        if (liege != "no_liege") {
+            out.println (tab+"liege="+"k_"+liege);
+            out.println (tab+"de_jure_liege="+"k_"+liege);
```
Same as above

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String provinceCreation(String ckProv, String ckCult, String ckRel, String Directory, String landType, 
```
The return type isn't used, so make this return void

Idhrendur replied:
Directory -> modDirectory

Idhrendur replied:
provinceCreation -> createProvince

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    String name, String gov, String pops, String[] bList, int id) throws IOException
+    {
+
+        String tab = "	";
```
As elsewhere, delete this and just put "\t" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    {
+
+        String tab = "	";
+        String VM = "\\"; 
```
As elsewhere, delete this and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String tab = "	";
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        Directory = Directory + VM + "history" + VM + "provinces";
```
Directory -> provincesDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    }
+
+    public static String provinceCreation(String ckProv, String ckCult, String ckRel, String Directory, String landType, 
+    String name, String gov, String pops, String[] bList, int id) throws IOException
```
bList -> baronyList

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        if (gov.equals ("tribal_federation") || gov.equals ("tribal_kingdom") || gov.equals ("tribal_chiefdom")) {
+            holding1 = "tribal";
+            holding2 = "tribal";
```
Are multiple tribal holdings in a country valid? I can only think of cases where there's one.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        }
+        else {
+            out.println (landType);
+        }
```
```
if (!landType.equals ("plains")) { 
  out.println (landType);
}
```

Or if its acceptable, print plains regardless just for simpler logic.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        name = name.replace(' ','_');
+
+        if (id == 103) { //Leon in Brittany and Spain have the same name in definition.csv
```
Instead of a hard-coded province id, have a private static final member variable in the class giving it a name:
```
private static final int FRENCH_LEON_ID = 103;
```

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+            out.println (landType);
+        }
+
+        if (id == 23) { //Stonehenge
```
As before, have a class variable for the province id

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+            out.println ("}");
+        }
+
+        if (id == 800) { //The great pyramids of Giza
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ckProv;
+    }
+
+    public static String ctitleCreation(String name, String irKING, String Directory, int id) throws IOException
```
The return value isn't used, so just return void

Idhrendur replied:
ctitleCreation -> createCountTitle

Idhrendur replied:
Directory -> modDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String religionOutput(String irRel) throws IOException
```
religionOutput -> outputReligion

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return 0;
+    }
+
+    public static String cultureOutput(String irCulture) throws IOException
```
cultureOutput -> outputCulture

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    {
+
+        btitleCreation(name,Directory,id);
+        String VM = "\\"; 
```
As elsewhere, delete this and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        btitleCreation(name,Directory,id);
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
```
As elsewhere, delete this and just put "\"" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        name = name.replace(' ','_');
+
+        if (id == 103) { //Leon in Brittany and Spain have the same name in definition.csv
```
As before, use a private static final member variable instead of a hardcoded province number

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+            name = "french_leon";  
+        }
+
+        Directory = Directory + VM + "history" + VM + "titles";
```
Directory -> titlesDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        Directory = Directory + VM + "history" + VM + "titles";
+
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
```
Only used for the return, so delete it.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        FileOutputStream fileOut= new FileOutputStream(Directory + VM + "c_" + name + ".txt");
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
```
Not used, so delete.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String date2 = "1066.9.15";
+
+        out.println (date1+"={");
+        out.println ("	holder="+irKING100);
```
\t instead of using the tab directly

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println ();
+
+        out.println (date2+"={");
+        out.println ("	holder="+irKING);
```
ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String characterCreation(String irKING, String cult, String rel, String age, String name, String dynasty,
```
The return value is never used, so make this return void.

Idhrendur replied:
cult -> culture

Idhrendur replied:
rel -> religion

Idhrendur replied:
characterCreation -> createCharacter

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    String mother,String Directory) throws IOException
+    {
+
+        String VM = "\\"; 
```
As elsewhere, drop this and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        String tab = "	";
```
As elsewhere, drop this and put "\t" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        String tab = "	";
+        char quote = '"';
```
As elsewhere, drop this and put "\"" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        VM = VM.substring(0);
+        String tab = "	";
+        char quote = '"';
+        String[] spouseInfo;
```
Declare this below, where it's actually used.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String tab = "	";
+        char quote = '"';
+        String[] spouseInfo;
+        String[] childInfo;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        char quote = '"';
+        String[] spouseInfo;
+        String[] childInfo;
+        int childCount;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String[] childInfo;
+        int childCount;
+        String spouse1066 = Integer.toString( 1000000 + Integer.parseInt(spouse));
+        String child1066;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String spouse1066 = Integer.toString( 1000000 + Integer.parseInt(spouse));
+        String child1066;
+
+        int hasFather = 0;
```
Use a boolean instead

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String child1066;
+
+        int hasFather = 0;
+        int hasMother = 0;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String father100 = "9" + father;
+        String mother100 = "9" + mother;
+
+        String dead = "no";
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        }
+
+        int aq4 = 0;
```
Declare this right before the loop where it's used.

Idhrendur replied:
aq4 -> childNum

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        }
+
+        if (children != "0") {
+            childCount = 1;
```
childCount -> totalChildren

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+                spouseInfo[4],  spouseInfo[8],  martial,  zeal,  charisma,  finesse,  "0",  "0", tempFile,"q",  "q", Directory);
+        }
+
+        if (children != "0") {
```
```
if (!children.equals("0")) {
```

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+                aq4 = aq4 + 1;
+            }
+            aq4 = 0;
```
No need to reset, its no longer used.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+    public static String characterCreation(String irKING, String cult, String rel, String age, String name, String dynasty,
+    String sex, String traits, String martial, String zeal, String charisma, String finesse, String spouse, String children,String tempFile,String father,
+    String mother,String Directory) throws IOException
```
Directory -> modDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        int numAge = Integer.parseInt(age);
+
+        
+        Directory = Directory + VM + "history" + VM + "characters";
```
Directory -> charactersDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        if (children != "0") {
+            childCount = 1;
+            try {
+                if (children.split(" ")[1] != null) {
```
Start this block by saving the results of children.split(" ") to a variable, and reference it instead of repeatedly redoing the split .

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+            }
+
+            while (aq4 < childCount) {
```
Use a for loop:
```
for (int i = 0; i < childCount; i++) {
```
Then drop aq4 entirely

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+            String[] traitList = traits.split(" ");
+
+            try {
+                while (aqq < 99) {
```
Use a for loop:
```
for (String trait: traitList) {
```

Then drop aqq entirely (you can declare it below when it become relevant in another context).

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+                        tempTrait = traitList[aqq];
+                    }
+
+                    tempTrait = Importer.importCultList("charTraitConverter.txt",tempTrait)[1];   
```
tempTrait -> convertedTrait

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String tempTrait = "a";
+        int aqq = 0;
+        int aq2 = 0;
```
aq2 -> numConvertedTraits

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+                    tempTrait = Importer.importCultList("charTraitConverter.txt",tempTrait)[1];   
+                    if (tempTrait != "99999") {
+                        convTraitList.add (tempTrait);
+                        aq2 = aq2 + 1;
```
convertedTraits += 1;

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+                    }
+                    aqq = aqq + 1;
+                }
+            }catch (java.lang.ArrayIndexOutOfBoundsException exception) {
```
Once using a for loop, this is no longer necessary

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        }
+
+        aqq = 0;
```
Remove this line, you can use a for loop below and only define the tracking variable within that scope.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        FileOutputStream fileOut= new FileOutputStream(Directory + VM + name + "k_" + irKING + ".txt");
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
```
Not used, delete.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        int flag = 0;
+
+        String date1 = "100.1.1";
```
Use the member variable constant.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        int flag = 0;
+
+        String date1 = "100.1.1";
+        String date2 = "1066.9.15";
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        if (traits != "q") {
+
+            while (aqq < aq2) {
```
Use a for loop:
```
for (String trait: convTraitList) {
```
Then you no longer need aqq, and I think you won't need aqq2 either.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String birthday2 = Integer.toString(birthdayNum2)+".9.15";
+        String birthday = Integer.toString(birthdayNum)+".1.1";
+
+        //Military/Martial Charisma/Stewardship Zeal/Learning
```
Redundant comment, delete.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        if (sex != "69") { //if a character is dynamically generated or not
+            out.println (tab+"disallow_random_traits = yes");    
+        }
+        //out.println (tab+"disallow_random_traits = yes");
```
Delete dead code

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        out.println (tab+"culture="+quote+cult+quote);
+
+        if (traits != "q") {
+            while (aqq < aq2) {
```
Use a for loop:
```
for (String trait: convTraitList) {
```

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return irKING;
+    }
+
+    public static String dynastyCreation(String name, String id, String Directory) throws IOException
```
The return value isn't used, so make this return void

Idhrendur replied:
Directory -> modDirectory

Idhrendur replied:
dynastyCreation -> createDynasty

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    public static String dynastyCreation(String name, String id, String Directory) throws IOException
+    {
+
+        String VM = "\\"; 
```
As elsewhere, remove this and just put "/" in the strings

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
```
As elsewhere, remove this and just put "\"" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
+        String tab = "	";
```
As elsewhere, remove this and just put "\t" in the strings

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        char VMq = '"';
+        String tab = "	";
+
+        Directory = Directory + VM + "common" + VM + "dynasties";
```
Directory -> dynastiesDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String tab = "	";
+
+        Directory = Directory + VM + "common" + VM + "dynasties";
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
```
This isn't used, so remove

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        FileOutputStream fileOut= new FileOutputStream(Directory + VM + "c_" + id + ".txt");
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        int flag = 0;
+
+        String date1 = "100.1.1";
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        int flag = 0;
+
+        String date1 = "100.1.1";
+        String date2 = "1066.9.15";
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String btitleCreation(String name, String Directory, int id) throws IOException
```
The return value isn't used, so make this function return void.

Idhrendur replied:
btitleCreation -> createBaronyTitle

Idhrendur replied:
Directory -> modDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    public static String btitleCreation(String name, String Directory, int id) throws IOException
+    {
+
+        String VM = "\\"; 
```
As elsewhere, drop this and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
```
As elsewhere, drop this and just put "\"" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
+        String tab = "	";
```
As elsewhere, drop this and just put "\t" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        VM = VM.substring(0);
+        char VMq = '"';
+        String tab = "	";
+        String oldName = name;
```
oldName isn't used anywhere, so drop it.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        char VMq = '"';
+        String tab = "	";
+        String oldName = name;
+        name = name.toLowerCase();
```
name -> cleanName or something of the sort (make sure to update the place where it's used).

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        name = name.toLowerCase();
+        name = name.replace(' ','_');
+
+        if (id == 103) { //Leon in Brittany and Spain have the same name in definition.csv
```
Use the constant created before.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+            name = "french_leon";  
+        }
+
+        Directory = Directory + VM + "common" + VM + "landed_titles";
```
Directory -> landedTitlesDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        Directory = Directory + VM + "common" + VM + "landed_titles";
+
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
```
This isn't used, so delete

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        FileOutputStream fileOut= new FileOutputStream(Directory + VM + "b_" + name + ".txt");
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int flag = 0;
```
Ditto

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String localizationCreation(String[] name, String title, String Directory) throws IOException
```
The return type isn't used, so make this function return void

Idhrendur replied:
localizationCreation -> createLocalization

Idhrendur replied:
Directory -> modDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    public static String localizationCreation(String[] name, String title, String Directory) throws IOException
+    {
+
+        String VM = "\\"; 
```
As elsewhere, drop this and just put "/" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
```
As elsewhere, drop this and just put "\"" in the strings.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        oldFile = Importer.importModLocalisation(Directory);
+
+        Directory = Directory + VM + "localisation";
```
Directory -> localisationsDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        oldFile = Importer.importModLocalisation(Directory);
+
+        Directory = Directory + VM + "localisation";
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
```
Not used, so delete

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        try {
+
+            while (flag == 0) {
```
It looks like you're just copying the original file with no modifications. Use java.nio.Files.copy to copy it, and delete the Importer.importModLocalisation function

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        return ck2CultureInfo;
+    }
+
+    public static String localizationBlankFile(String Directory) throws IOException
```
The return value isn't sued, so make this function return void

Idhrendur replied:
localizationBlankFile -> createBlankLocalisationFile

Idhrendur replied:
Directory -> modDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+    public static String localizationBlankFile(String Directory) throws IOException
+    {
+
+        String VM = "\\"; 
```
As elsewhere, just use "/" in the strings or expressions themselves.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
```
As elsewhere, just use "\"" in the strings or expressions themselves.

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        VM = VM.substring(0);
+        char VMq = '"';
+
+        Directory = Directory + VM + "localisation";
```
Directory -> localisationDirectory

Idhrendur commented on ImperatorToCk2/Output.java:
```diff
@@ -0,0 +1,737 @@
+        char VMq = '"';
+
+        Directory = Directory + VM + "localisation";
+        String ck2CultureInfo ="a";   // blank default
```
Unused, so delete

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+ */
+public class Characters
+{
+    private int x;
```
Not used anywhere, delete

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+    public static String[] importChar (String name, String charID) throws IOException
+    {
+
+        String tab = "	";
```
As elsewhere, remove and use "\t" in the strings

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+    {
+
+        String tab = "	";
+        char quote = '"';
```
As elsewhere, remove and use "\"" in the strings

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        FileInputStream fileIn= new FileInputStream(name);
+        Scanner scnr= new Scanner(fileIn);
+
+        int flag = 0;
```
Define just before the loop where its used

Idhrendur replied:
Also, this should be a boolean

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        int aqq = 0;
+
+        boolean endOrNot = true;
```
Move just before the loop where its used

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        int flag = 0;
+
+        String keyWord = charID+"={";
```
Instead of a variable, just place the string directly where its used.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        int aqq = 0;
+
+        boolean endOrNot = true;
+        String vmm = scnr.nextLine();
```
This variable isn't really used. Drop it and assign where you really care. Or if you're dropping a line, assign it to a variable called unused.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        boolean endOrNot = true;
+        String vmm = scnr.nextLine();
+        String qaaa = vmm;
```
Declare this variable inside the loop where you use it.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        output[16] = "0"; //default for minor character with no family name
+
+        try {
+            while (endOrNot = true){
```
Have an initial loop that scans until you hit keyword:
```
String line = scnr.nextLine();
while (!line.equals(keyWord)) {
  line = scnr.nextLine();
}
```

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        try {
+            while (endOrNot = true){
+
+                qaaa = scnr.nextLine();
```
After adding the simpler loop, this in unnecessary.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+            while (endOrNot = true){
+
+                qaaa = scnr.nextLine();
+                if (qaaa.equals(keyWord)){
```
Ditto.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+                qaaa = scnr.nextLine();
+                if (qaaa.equals(keyWord)){
+                    endOrNot = false;
```
Ditto.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+                if (qaaa.equals(keyWord)){
+                    endOrNot = false;
+                    while (flag == 0) {
+                        qaaa = scnr.nextLine();
```
qaaa -> line

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+                        }
+                        else if (qaaa.split("=")[0].equals( tab+"traits" ) ) {
+                            System.out.println (qaaa); 
```
If you don't need to keep logging this, drop it.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+                            output[8] = qaaa.split("=")[1];
+
+                            output[8] = output[8].substring(2,output[8].length()-2);
+                            System.out.println (output[8]);
```
Ditto

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+                        }
+
+                        //popTotal
```
This comment doesn't seem to match the code. Drop it.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+                            output[3] = qaaa.split("=")[1];
+                        }
+
+                        //might be used or ignored
```
Drop comment

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        return output;
+    
+    }
+    public static String importAndConvDynasty (String dir, String IDnum, String backupName, String file) throws IOException
```
The return value isn't used, so return void.

Idhrendur replied:
importAndConvertDynasty

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+    public static String importAndConvDynasty (String dir, String IDnum, String backupName, String file) throws IOException
+    {
+
+        String tab = "	";
```
As elsewhere, drop this and just put "\t" in the strings.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        FileInputStream fileIn= new FileInputStream(file);
+        Scanner scnr= new Scanner(fileIn);
+
+        int flag = 0;
```
Define this just before the loop where it's used.

Idhrendur replied:
Also make it a boolean

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        int flag = 0;
+
+        String keyWord = tab+tab+IDnum;
```
Instead of a variable, put this in the expression where it's used.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        String keyWord = tab+tab+IDnum;
+
+        int aqq = 0;
```
Unused, delete

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        String keyWord = tab+tab+IDnum;
+
+        int aqq = 0;
+        boolean endOrNot = true;
```
Move just before the loop where it's used.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+        int aqq = 0;
+        boolean endOrNot = true;
+        String vmm = scnr.nextLine();
```
Unnecessary variable, drop and just assign to the variable you want.

Idhrendur replied:
Also, move this into the loop where you want it. Unless you're skipping a single line, in which case assign it to a variable called unused.

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+
+      
+        try {
+            while (endOrNot = true){
```
Looks like this just scans until it runs out of input. Make it
```
while (true) {
```

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+        try {
+            while (endOrNot = true){
+
+                qaaa = scnr.nextLine();
```
qaaa -> line

Idhrendur commented on ImperatorToCk2/Characters.java:
```diff
@@ -0,0 +1,208 @@
+                qaaa = scnr.nextLine();
+                if (qaaa.split("=")[0].equals(keyWord)){
+                    endOrNot = false;
+                    while (flag == 0) {
```
You can make this
```
while (true) { 
```
Then when you're ready to stop, instead of setting the flag, just give the statement
```
break;
```

DrDudelsack committed:
Bugfixes, localization improvements, and mapping

Removed most unused variables from classes, fully mapped out all cultures, religions, and provinces. Added localization support for popular Imperator:Rome mods, fixed localization for countries which use a dynasty name in their title (Seleukids, Antigonids, and Maurya in the start date), improved localization for dynamic countries with an area name as their name, and added localization for governorship titles. 

Fixed a major bug in Output.characterCreation causing children of already converted characters who had jobs not to have any relationship to their mother and father due to being converted twice by creating a list of all previously converted characters which would be parsed in as a parameter in the method, which is updated for each character converted.

DrDudelsack committed:
Added Sumerian/Mesopotamian

DrDudelsack committed:
Added cultures

Added cultures for the Anatolian and Latin groups from Imperator:Rome, moved Basque to the Pre-Indo European group, and removed previously added unused cultures
